lib/puppet/provider/grafana_dashboard/grafana.rb

```diff
@@ -160,10 +161,12 @@ def create
   end
 
   def destroy
-    response = send_request('DELETE', format('%s/dashboards/uid/%s', resource[:grafana_api_path], slug))
+    db = dashboards.find { |x| x['title'] == slug }
```

Comment:
Here we are destroying the dashboard based on the slug, but `find_dashboard` function uses `resource[:title]`.
Probably slug is more suitable, but we are creating dashboards with `resource[:title]` as title. This can cause problems in case `title` contains dashes and spaces.

Reply:
Good idea, fixed that